pkg/scheduler/apis/config/types_test.go

```diff
@@ -224,59 +224,50 @@ func TestPluginsApply(t *testing.T) {
 				},
 			},
 			defaultPlugins: &Plugins{
-				Filter: &PluginSet{
+				Filter: PluginSet{
 					Enabled: []Plugin{
 						{Name: "DefaultPlugin1"},
 						{Name: "DefaultPlugin2"},
 					},
 				},
 			},
 			expectedPlugins: &Plugins{
-				QueueSort: &PluginSet{Enabled: []Plugin{}},
-				PreFilter: &PluginSet{Enabled: []Plugin{}},
-				Filter: &PluginSet{
+				QueueSort: PluginSet{Enabled: []Plugin{}},
+				PreFilter: PluginSet{Enabled: []Plugin{}},
+				Filter: PluginSet{
 					Enabled: []Plugin{
 						{Name: "DefaultPlugin2"},
 						{Name: "DefaultPlugin1"},
 					},
 				},
-				PostFilter: &PluginSet{Enabled: []Plugin{}},
-				PreScore:   &PluginSet{Enabled: []Plugin{}},
-				Score:      &PluginSet{Enabled: []Plugin{}},
-				Reserve:    &PluginSet{Enabled: []Plugin{}},
-				Permit:     &PluginSet{Enabled: []Plugin{}},
-				PreBind:    &PluginSet{Enabled: []Plugin{}},
-				Bind:       &PluginSet{Enabled: []Plugin{}},
-				PostBind:   &PluginSet{Enabled: []Plugin{}},
+				PostFilter: PluginSet{Enabled: []Plugin{}},
```

Comment:
Actually, the objective should be to get rid of these lines.

They should be something like: `PostFilter PluginSet{Enabled: nil, Disabled: nil}`, which you can actually just not remove.

Reply:
Wondering what to do here

Reply:
Uhmmm could it be that `append(nil, (nil)...)` in `appendPluginSet` is creating a pointer? You could experiment in https://play.golang.org/

Reply:
I have done some changes..PTAL..The test now passes.